### PR TITLE
Remove friends declaration in Submap.

### PR DESCRIPTION
**Pull request by SirVer**

### CODE REVIEW

**Comment by SirVer** [comment]
Based on #348, which needs to be merged first.

**Comment by SirVer** [comment]
Rebased and ready for review.

File: cartographer/mapping_2d/submaps.cc [hunk 1]
```diff
@@ -161,11 +176,10 @@ ActiveSubmaps::ActiveSubmaps(const proto::SubmapsOptions& options)
 
 void ActiveSubmaps::InsertRangeData(const sensor::RangeData& range_data) {
   for (auto& submap : submaps_) {
-    CHECK(!submap->finished_);
-    range_data_inserter_.Insert(range_data, &submap->probability_grid_);
-    ++submap->num_range_data_;
+    submap->InsertRangeData(range_data, range_data_inserter_);
   }
-  if (submaps_.back()->num_range_data_ == options_.num_range_data()) {
+  if (submaps_.back()->num_range_data() ==
```
Comment (wohe) [inline cartographer/mapping_2d/submaps.cc]: Change `num_range_data()` to return an `int` instead?

File: cartographer/mapping_2d/submaps.cc [hunk 2]
```diff
@@ -151,6 +151,21 @@ void Submap::ToResponseProto(
...
+void Submap::InsertRangeData(const sensor::RangeData& range_data,
+                             const RangeDataInserter& range_data_inserter) {
+  CHECK(!finished_);
+  range_data_inserter.Insert(range_data, &probability_grid_);
+  ++num_range_data_;
+}
+
+void Submap::Finish() {
+  CHECK(!finished_);
+  // Crop the finished Submap before inserting a new Submap to reduce peak
```
Comment (wohe) [inline cartographer/mapping_2d/submaps.cc]: I think this comment is not understandable here. Move next to the call to `FinishSubmap()` in `AddSubmap()`?

File: cartographer/mapping_3d/submaps.cc [hunk 3]
```diff
@@ -381,27 +401,19 @@ void ActiveSubmaps::InsertRangeData(
...
-                                  options_.high_resolution_max_range()),
-        &submap->high_resolution_hybrid_grid_);
-    range_data_inserter_.Insert(transformed_range_data,
-                                &submap->low_resolution_hybrid_grid_);
-    ++submap->num_range_data_;
+    submap->InsertRangeData(range_data, range_data_inserter_,
+                            options_.high_resolution_max_range());
   }
-  if (submaps_.back()->num_range_data_ == options_.num_range_data()) {
+  if (submaps_.back()->num_range_data() ==
```
Comment (wohe) [inline cartographer/mapping_3d/submaps.cc]: Same here: Make `num_range_data()` return the `int`?

**Pull request merged**